query-engine/driver-adapters/src/send_future.rs

```diff
@@ -0,0 +1,27 @@
+use futures::Future;
+
+// Allow asynchronous futures to be sent safely across threads, solving the following error:
+//
+//  ```text
+// future cannot be sent between threads safely
+// the trait `Send` is not implemented for `dyn Future<Output = std::result::Result<u32, js_sys::Error>>`.
+// ```
+//
+// See: https://github.com/rustwasm/wasm-bindgen/issues/2409#issuecomment-820750943
```

Comment:
TODO:
- make it `///` for docs purposes
- remove "sent safely"
- highlight this is about `wasm-bindgen` limitations
- remind that we should probably don't require `Send` future for Wasm in the first place (this requires `#[async_trait(?)]` in e.g. the `Queryable` trait definition)